index_review.js

```diff
@@ -0,0 +1,257 @@
+// Node packages being used
+const express = require("express");
+const bodyParser = require("body-parser"); // Parses JSON bodies
+const cookieParser = require("cookie-parser");
+const app = express().use(bodyParser.text());
+const port = process.env.PORT || 3000;
+const axios = require("axios"); // Helps by sending HTTP for us
+
+// Configuration of our server
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+app.use(cookieParser());
+
+// Productboard and GitLab information here
+const pbIntegrationID = process.env.PB_INTEGRATION_ID; // Productboard plugin integration ID here
+const productboardToken = "Bearer " + process.env.PB_TOKEN; // Productboard API token here to authorize requests
+const gitlabProjectID = process.env.GITLAB_PROJECT_ID; // GitLab Project ID
+const gitlabToken = process.env.GITLAB_TOKEN; // Gitlab Access token
+
+// Initial route to confirm app is running on Heroku
+app.get("/", (req, res) => {
+	res.send(
+		"This is a Heroku server hosting our Productboard <> GitLab integration"
+	);
+});
+
+// Route to authenticate plugin connection
+app.get("/plugin", (req, res) => {
+	res.setHeader("Content-type", "text/plain");
+	res.status(200).send(res.req.query.validationToken);
+});
+
+// Optional route if webhooks from Productboard are needed to support a 2-way sync
+app.get("/productboard-webhook", async (req, res) => {
+	res.setHeader("Content-type", "text/plain");
+	res.status(200).send(res.req.query.validationToken);
+});
+
+// Endpoint where POST requests from Productboard plugin will be sent
+app.post("/plugin", async (req, res) => {
+	// Gather information about the feature that is sending over request
+	const pbFeatureID = res.req.body.data.feature.id;
+	const pbIntegrationID = res.req.body.data.integrationId;
+	const pbFeatureLink = res.req.body.data.feature.links.html;
+
+	console.log("Productboard trigger is:", req.body.data.trigger);
+
+	// Determine action on button trigger. Can be push, dismiss, or unlink.
+	if (req.body.data.trigger === "button.push") {
+		res.json({
+			data: {
+				connection: {
+					state: "progress",
+				},
+			},
+		});
+
+		// Setup request to access information of feature being pushed
+		const pbFeatureInfo = {
```

Comment:
Might be worth to extract `sendRequestToProductboard` and `sendRequestToGitlab` methods with predefined base urls and headers :) It would help readability a bit.